crates/hir/src/semantics.rs

```diff
@@ -129,6 +129,7 @@ pub struct SemanticsImpl<'db> {
     cache: RefCell<FxHashMap<SyntaxNode, HirFileId>>,
     // MacroCall to its expansion's HirFileId cache
     macro_call_cache: RefCell<FxHashMap<InFile<ast::MacroCall>, HirFileId>>,
+    analyze_impl_cache: RefCell<FxHashMap<(SyntaxNode, Option<TextSize>, bool), SourceAnalyzer>>,
```

Comment:
I have doubts that this caching will have any effect as the key is rather unique here(that is the `Option<TextSize>` makes this rather pointless I believe), though I may be wrong

Reply:
Yeah, based on my (unscientific) testing, this doesn't seem to make any noticeable improvements.